streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPeek.java

```diff
@@ -35,10 +39,35 @@ public KStreamPeek(final ForeachAction<K, V> action) {
     }
 
     private class KStreamPeekProcessor extends AbstractProcessor<K, V> {
+        
+        @Override
+        public void init(ProcessorContext context) {
+            super.init(context);
+            if (action instanceof PrintForeachAction) {
+                ((PrintForeachAction) action).setContext(context);
```

Comment:
You could to the case only once, and introduce a new variable. This would increase readability.